Odin lang filetype/syntax/indent plugin

habamax opened the pull request:
Vim has Odin filetype detection but misses actual Odin filetype, syntax and indent.

Repo: https://github.com/habamax/vim-odin

![image](https://github.com/vim/vim/assets/234774/18670dca-c732-4e9e-973f-5a93f29a7852)

![image](https://github.com/vim/vim/assets/234774/c7ed8f29-fece-45f9-a948-9d16fe4ad1e7)

habamax committed:
odin-lang ft/syntax/indent

habamax committed:
Add Last Change

habamax committed:
fix long /* multiline comments */

Use 'syn sync ccomment'

habamax committed:
support nested block comments

as in:

    // Multi-line comments begin with /* and end with */. Multi-line comments can
    // also be nested (unlike in C):
    /*
            You can have any text or code here and
            have it be commented.
            /*
                    NOTE: comments can be nested!
            */
    */

dkearns commented on runtime/ftplugin/odin.vim:
```diff
@@ -0,0 +1,18 @@
+vim9script
+
+# Vim filetype plugin file
+# Language: odin
```
s/odin/Odin/

dkearns commented on runtime/ftplugin/odin.vim:
```diff
@@ -0,0 +1,18 @@
+      \ .. '| setlocal suffixesadd<'
+
+setlocal suffixesadd=.odin
+setlocal commentstring=//%s
```
Could we also set `comments` for formatting commands?

dkearns commented on runtime/syntax/odin.vim:
```diff
@@ -0,0 +1,103 @@
+syntax match odinTemplate "$\<\w\+\>"
+
+syntax region odinLineComment start=/\/\// end=/$/  contains=@Spell,odinTodo
+syntax region odinBlockComment start=/\v\/\*/ end=/\v\*\// contains=@Spell,odinTodo,odinBlockComment
```
The `\v` modifier isn't doing any work here.

dkearns reviewed (changes requested):
Thanks, I was just watching an interview with the ginger one.

You're probably aware but Vim9-script runtime files won't be picked up by Neovim.  They're not rewriting them in legacy script.

habamax committed:
update runtime files

- Change lang name to Odin
- add 'comments'
- remove redundant \v
- fix indent issue with labeled: for

habamax committed:
improve indent

habamax committed:
fix additional indent edge cases

habamax committed:
cover more indent edge cases

habamax committed:
More indent edge cases
